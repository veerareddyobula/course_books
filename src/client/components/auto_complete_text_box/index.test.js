import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import AutoCompleteTextBox from './index';

const mockStore = configureStore([]);

describe('AutoCompleteTextBox Component', () => {
  let store;
  let component;
  beforeEach(() => {
    store = mockStore({
      fetchTitlesList: () => true,
      booksTitlesList: []
    });
    const changeHandler = () => true;
    store.dispatch = jest.fn();
    component = renderer.create(
      <Provider store={store}>
        <AutoCompleteTextBox val="Fr" onChangeHandler={changeHandler} />
      </Provider>
    );
  });
  it('AutoCompleteTextBox Component Snapshot', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});
