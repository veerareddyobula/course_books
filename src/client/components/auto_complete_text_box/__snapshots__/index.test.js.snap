// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AutoCompleteTextBox Component AutoCompleteTextBox Component Snapshot 1`] = `
<div
  className="auto-complete"
>
  <input
    onChange={[Function]}
    placeholder="Search books"
    type="text"
    value="Fr"
  />
  <div>
    <ul />
  </div>
</div>
`;
